### PR TITLE
Add HTTP auth for downloads from TanDEM-X

### PR DESCRIPTION
**Pull request by TimoRoth**

### CODE REVIEW

**Comment by TimoRoth** [comment]
Still WIP, but Travis won't run on Drafts.

**Comment by TimoRoth** [comment]
Passes the test and generally works fine for me now. @fmaussion 
Might still do one or two minor changes, so please don't merge just yet.

File: oggm/utils/_downloads.py [hunk 1]
```diff
@@ -526,18 +532,25 @@ def _download_tandem_file_unlocked(zone):
...
+    with open(tdmauthfile, 'r') as f:
+        tdmuser = f.readline().strip()
+        tdmpass = f.readline().strip()
+    if not tdmuser or not tdmpass:
+        raise DownloadCredentialsMissingException(
+            'Could not read credentials from ' + tdmauthfile)
+
     # Did we download it yet?
-    scpfile = ('/home/data/download/tandemx-90m.dlr.de/90mdem/DEM/'
+    wwwfile = ('https://download.geoservice.dlr.de/TDM90/files/'
```
Comment (fmaussion) [inline oggm/utils/_downloads.py]: This change means we will have to update the paths on cluster, right?

File: oggm/utils/_downloads.py [hunk 2]
```diff
@@ -526,18 +532,25 @@ def _download_tandem_file_unlocked(zone):
     if os.path.exists(outpath):
         return outpath
 
+    # Grab auth parameters
+    tdmauthfile = os.path.expanduser('~/.tdmdem90.creds')
+    if not os.path.isfile(tdmauthfile):
+        raise DownloadCredentialsMissingException(
+            tdmauthfile + ' does not exist.')
```
Comment (fmaussion) [inline oggm/utils/_downloads.py]: Add instructions on how to generate this file in the error message.

File: oggm/utils/_downloads.py (hunk 1 above)
Comment (TimoRoth) [reply oggm/utils/_downloads.py]: No, that's where they already are.

File: oggm/utils/_downloads.py (hunk 2 above)
Comment (TimoRoth) [reply oggm/utils/_downloads.py]: You think what I added is enough? I don't want to write too many instructions into an exception message, but what's there now should be enough to allow figuring things out.

**Pull request merged**